packages/api/src/controllers/managementPlanController.js

```diff
@@ -14,23 +14,107 @@
  */
 
 import ManagementPlanModel from '../models/managementPlanModel.js';
+import ManagementPlanGroup from '../models/managementPlanGroupModel.js';
 import CropManagementPlanModel from '../models/cropManagementPlanModel.js';
 import ManagementTasksModel from '../models/managementTasksModel.js';
 import TaskModel from '../models/taskModel.js';
 import TaskTypeModel from '../models/taskTypeModel.js';
 import FieldWorkTypeModel from '../models/fieldWorkTypeModel.js';
 import TransplantTaskModel from '../models/transplantTaskModel.js';
 import PlantTaskModel from '../models/plantTaskModel.js';
-import { raw } from 'objection';
-import lodash from 'lodash';
+import UserFarmModel from '../models/userFarmModel.js';
+import objection, { raw } from 'objection';
+import _pick from 'lodash/pick.js';
 import { sendTaskNotification, TaskNotificationTypes } from './taskController.js';
+import {
+  getDatesFromManagementPlanGraph,
+  getManagementPlanGroupTemplateGraph,
+  getFormattedManagementPlanData,
+} from '../util/copyCropPlan.js';
+import { getSortedDates } from '../util/util.js';
+const { transaction, Model } = objection;
 
 const managementPlanController = {
+  repeatManagementPlan() {
+    return async (req, res) => {
+      const trx = await transaction.start(Model.knex());
+      const { start_dates, management_plan_id, repeat_details } = req.body;
+      try {
+        if (!start_dates?.length > 0 || !management_plan_id || !repeat_details?.crop_plan_name) {
+          throw 'Insufficient details to copy crop plan';
+        }
+        const createdByUser = req.auth.user_id;
+
+        // Get source management plan entire graph acting as a template
+        const managementPlanGraph = await ManagementPlanModel.query(trx)
+          .where('management_plan_id', management_plan_id)
+          .withGraphFetched(
+            'crop_management_plan.[planting_management_plans.[managementTasks.[task.[pest_control_task, irrigation_task, scouting_task, soil_task, soil_amendment_task, field_work_task, harvest_task, cleaning_task, locationTasks]], plant_task.[task.[locationTasks]], transplant_task.[task.[locationTasks]], bed_method, container_method, broadcast_method, row_method]]',
+          )
+          .first();
+
```

Comment:
This is missing a filter on deleted tasks which is needed according to ticket scope. But I'm sure why the created graph doesn't have the deleted property within the task objects. Do you know?